spec/Appendix B -- Grammar Summary.md

```diff
@@ -172,3 +173,57 @@ NonNullType :
 Directives : Directive+
 
 Directive : @ Name Arguments?
+
+TypeSystemDefinition :
+  - SchemaDefinition
+  - TypeDefinition
+  - TypeExtensionDefinition
+  - DirectiveDefinition
+
+SchemaDefinition : schema Directives? { OperationTypeDefinition+ }
+
+OperationTypeDefinition : OperationType : NamedType
+
+TypeDefinition :
+  - ScalarTypeDefinition
+  - ObjectTypeDefinition
+  - InterfaceTypeDefinition
+  - UnionTypeDefinition
+  - EnumTypeDefinition
+  - InputObjectTypeDefinition
+
+ScalarTypeDefinition : scalar Name Directives?
+
+ObjectTypeDefinition : type Name ImplementsInterfaces? Directives? { FieldDefinition+ }
```

Comment:
I have noticed that some implementations (like `GraphQL Ruby`) do emit objects in the IDL that lack fields, such as `type Foo {}`. Will that be taken into consideration here? (Making this `FieldDefinition*`.)

Reply:
Thanks for identifying this! While this would result in an invalid final schema - it's still useful to parse as an intermediate schema - I'll correct for this

Reply:
@leebyron Could you help me with how it's useful to parse something knowingly invalid? Or is that in the expectation of an `extends`?

Reply:
To resolve this, rather than allow parsing `type Foo {}` which you correctly point out would be invalid, I've changed the definition of extends to explicitly support a case to extend a type without adding fields.